components/NoteEditor/Editor/Editor.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import dynamic from 'next/dynamic';
+
+import shortid from 'shortid';
+import styles from './Editor.module.scss';
+import Heading from './Heading';
+
+const Section = dynamic(() => import('./Section'), { ssr: false });
+
+function formatDate(dateObj) {
+  return new Intl.DateTimeFormat('en-US', { month: 'short', day: 'numeric' }).format(dateObj);
+}
+
+const formatedDate = formatDate(new Date()).split(' ');
+
+export default function Editor() {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [sections, setSections] = useState([{ id: shortid.generate(), keyword: '', content: '' }]);
+  const [summary, setSummary] = useState('');
+
+  const handleAddSection = () => {
+    const newSection = { id: shortid.generate(), keyword: '', content: '' };
+    setSections(sections.concat(newSection));
```

Comment:
Concat is pretty slow, you could use the spread operator if you're going the functional route or else just use push. Push is the fastest since it doesnt create a new array but changes the array inplace. 
```suggestion
    setSections([...sections, newSection]);
```

Reply:
You can't use functions that change the state directly. spread operator sounds good to me

Reply:
👍